### PR TITLE
[DEV-3627] Minor edits to DABS ETL

### PR DESCRIPTION
**Pull request by tony-sappe**
**Description:**
DEV-3627 was for investigating and correcting "missing" DABS data in USAspending. The investigation didn't turn up anything definitive, nothing in the current ETL code nor past history seemed to be a concern. My suspicion is currently focused on Broker's reference TAS data & FKs to those records, but it doesn't impact USAspending's ETL so an investigation into Broker data can wait.

The main value from this work is `usaspending_api/etl/management/sql/dev3627_compare_dabs_row_counts.sql`. The steps to fully utilize the SQL is the following:

1. Run `dev3627_compare_dabs_row_counts.sql` to find a list of submissions which have count discrepancies
2. Run `load_multiple_submissions --submission-ids <ids>` with the IDs from step 1
3. Re-run `dev3627_compare_dabs_row_counts.sql` to confirm no discrepancies remain

**Technical details:**
The code edits are pretty minor. Mostly logging changes or removing cruft. There was one issue of the File B select not obtaining `row_number` in the original code (which might not have been a concern with production data) and was ultimately fixed by removing any need for `row_number` since it wasn't being beneficially used anyway. 

`dev3627_compare_dabs_row_counts.sql` can be run anytime to compare the counts of account data in each submission between Broker and USAspending. It is a lengthly run (20+ minutes) so it isn't recommended to add to the nightly pipeline without a solid reason.

**Requirements for PR merge:**

1. [ ] Unit & integration tests updated
2. [x] API documentation updated (N/A)
3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. [x] Matview impact assessment completed (N/A)
5. [x] Frontend impact assessment completed (N/A)
6. [x] Data validation completed
7. [ ] Appropriate Operations ticket(s) created
8. [x] Jira Ticket [DEV-3627](https://federal-spending-transparency.atlassian.net/browse/DEV-3627):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected SQL script
    - [x] Before / After data comparison

**Area for explaining above N/A when needed:**
```
No impact to materialized views or API
```

### CODE REVIEW

File: usaspending_api/etl/submission_loader_helpers/treasury_appropriation_account.py [hunk 1]
```diff
@@ -28,10 +28,7 @@ def bulk_treasury_appropriation_account_tas_lookup(rows, db_cursor):
                     sub_account_code
             from    tas_lookup
             where   account_num in %s
-                    and (
-                        financial_indicator2 != 'F'
-                        or financial_indicator2 is null
-                    )
+                    and financial_indicator2 IS DISTINCT FROM 'F'
```
Comment (sethstoudenmier) [inline usaspending_api/etl/submission_loader_helpers/treasury_appropriation_account.py:31]: Do we no longer need to filter out `NULL`?

File: usaspending_api/submissions/management/commands/rm_submission.py [hunk 2]
```diff
@@ -46,10 +44,22 @@ def signal_handler(signal, frame):
 
         deleted_stats = submission.delete()
 
-        self.logger.info("Finished deletions.")
+        models = {
```
Comment (sethstoudenmier) [inline usaspending_api/submissions/management/commands/rm_submission.py:47]: This is a really nice addition!

File: usaspending_api/etl/management/sql/dev3627_compare_dabs_row_counts.sql [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    GROUP BY submission_id
+), file_b AS (
+    SELECT submission_id, COUNT(fapaoc.*) AS count
+    FROM submission_attributes sa
+    LEFT OUTER JOIN financial_accounts_by_program_activity_object_class AS fapaoc USING (submission_id)
+    GROUP BY submission_id
+), file_c AS (
+    SELECT sa.submission_id, COUNT(faba.*) AS count
+    FROM submission_attributes sa
+    LEFT OUTER JOIN financial_accounts_by_awards faba USING (submission_id)
```
Comment (sethstoudenmier) [inline usaspending_api/etl/management/sql/dev3627_compare_dabs_row_counts.sql:14]: Really small, but would be nice to have `AS` in this line as well for consistency.

**Review by sethstoudenmier — APPROVED** [review]
LGTM. Minor thing I saw on the SQL for consistency, but makes no difference for functionality.

File: usaspending_api/etl/management/sql/dev3627_compare_dabs_row_counts.sql (hunk 3 above)
Comment (tony-sappe) [reply usaspending_api/etl/management/sql/dev3627_compare_dabs_row_counts.sql]: Done, thanks

File: usaspending_api/submissions/management/commands/rm_submission.py (hunk 2 above)
Comment (tony-sappe) [reply usaspending_api/submissions/management/commands/rm_submission.py:47]: Glad my own discomfort with the logging was felt by others

**Review by sethstoudenmier — APPROVED** [review]

**Pull request merged**